### PR TITLE
Introduce OCI artifacts signatures

### LINKED ISSUE #65: Sign the OCI artifact with cosign
**Motivation**

This feature request is part of a work which aims to improve the security of the Falco artifacts supply chain.

**Feature**

Introduce a step in the release pipeline which signs the rules OCI artifact built, with [cosign](https://github.com/sigstore/cosign).

**Alternatives**

Use different tools, like [Notary](https://github.com/notaryproject/notation), that provide different format, even though supporting same storage type (OCI Artifact). 

**Additional context**

In detail, of signatures of Falco OCI artifacts, you can read [here](https://github.com/falcosecurity/falcoctl/issues/174#issuecomment-1499338972).

### PR DESCRIPTION
**Pull request by maxgio92**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area registry
/area build

**What this PR does / why we need it**:

This PR introduces the signature of the rules' OCI artifact, usign [cosign](https://github.com/sigstore/cosign).

**Which issue(s) this PR fixes**:

Fixes #65 

**Special notes for your reviewer**:

This is part of the work for securing the Falco supply chain. In detail of signatures of Falco OCI artifacts, you can read [here](https://github.com/falcosecurity/falcoctl/issues/174#issuecomment-1499338972).

### CODE REVIEW

**Comment by maxgio92** [comment]
/hold

This is a work in progress and the publish to S3-steps are missing.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -9,50 +9,59 @@ on:
...
           REGISTRY_USER: ${{ github.repository_owner }}
           REGISTRY_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           GITHUB_REPO_URL: ${{ github.server_url }}/${{ github.repository }}.git
 
         # uses OCI_REPO_PREFIX environment variable
-        run: build/registry/rules-registry push-to-oci registry.yaml ${{ github.ref_name }}
+        run: >-
+          echo "::set-output name=ARTIFACT_DIGEST::$(
+            build/registry/rules-registry push-to-oci registry.yaml ${{ github.ref_name }}
+          )"
```
Comment (LucaGuerra) [inline .github/workflows/release.yaml:53]: I believe `::set-output` is deprecated and we should use the GITHUB variables instead: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Comment (maxgio92) [reply .github/workflows/release.yaml:53]: Good catch @LucaGuerra thanks!

**Comment by LucaGuerra** [comment]
> This is a work in progress and the publish to S3-steps are missing.

Why? That step is embedded in the Go builder binary normally 💡 , was it disabled somewhere?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (maxgio92) [reply .github/workflows/release.yaml]: @LucaGuerra fixed it, PTAL, thanks

**Comment by LucaGuerra** [comment]
@maxgio92 do we want to merge plugins first and adapt this? I guess the two will use the same mechanism we discussed in the other PR!

**Comment by maxgio92** [comment]
Totally agree @LucaGuerra

**Comment by maxgio92** [comment]
Rebased on main branch

**Review by LucaGuerra — APPROVED** [review]
LGTM

**Comment by LucaGuerra** [comment]
@maxgio92 if this is ready you can remove the hold :)

**Pull request merged**